components/widgets/menu/menu_items/menu_start_trial.tsx

```diff
@@ -0,0 +1,54 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React from 'react';
+import {useIntl} from 'react-intl';
+import {useDispatch, useSelector} from 'react-redux';
+import {useHistory} from 'react-router-dom';
+
+import {getLicenseConfig} from 'mattermost-redux/actions/general';
+import {requestTrialLicense} from 'actions/admin_actions';
+import {GlobalState} from 'mattermost-redux/types/store';
+
+import './menu_item.scss';
+
+type Props = {
+    id: string;
+    show: boolean;
+}
+
+const MenuStartTrial = (props: Props): JSX.Element | null => {
+    const {formatMessage} = useIntl();
+    const history = useHistory();
+
+    const dispatch = useDispatch();
+    const stats = useSelector((state: GlobalState) => state.entities.admin.analytics);
+
+    const requestLicense = async () => {
+        let users = 0;
+        if (stats && (typeof stats.TOTAL_USERS === 'number')) {
+            users = stats.TOTAL_USERS;
+        }
+        const requestedUsers = Math.max(users, 30);
+        await dispatch(requestTrialLicense(requestedUsers, true, true, 'license'));
```

Comment:
Looks like this actually makes the request to start a trial, which surprises me as a user. I would expect to be taken to some sort of model or page with more details about the trial, so that I can understand what the terms are before I agree to them/confirm that I really mean to do this. It seems ambiguous in the [JIRA ticket](https://mattermost.atlassian.net/browse/MM-37978) what the actual expectations are. @esethna @marianunez , what should clicking on the link do?

Reply:
@neallred good question, I agree with your concerns. I'll leave that question to @jasonblais as he is likely more familiar with this ticket

Reply:
@neallred and @esethna , I agree. The alternative is to take them to admin console

Reply:
Agree, good note. I believe we had designed an intermediate screen that we planned to use in other "Start Trial" CTAs before. We could use it here. I'll try find it from the design team, and post back here.

(I'm not sure if this increases the scope of the work for this ticket, but we can discuss)

Reply:
Should we open this modal that will have a more intermediate step: https://www.figma.com/file/E2sh7Ny5wlyDhqpgbwowxm/Enterprise-changes?node-id=2%3A71322?

<img width="730" alt="Screen Shot 2021-08-24 at 5 00 46 PM" src="https://user-images.githubusercontent.com/936315/130689239-df163c89-9d93-4157-849a-013aebccf36c.png">

Reply:
https://user-images.githubusercontent.com/28563179/130783912-66a57070-55ea-40c0-ba7e-28881e96672f.mov

Reply:
Looks great to me :+1:

Reply:
@marianunez and @annelieseklein , I have added an intermediate flow. See screen recording above

Reply:
Yes, this modal was meant to be included in this step, thank you @marianunez @AGMETEOR ! Looks great!